### PR TITLE
Release 0.102.0

### LINKED ISSUE #3012: Update Collector to v0.102.0
v0.102.0 has been released, it's time for the Operator to upgrade to v0.102.0!

### PR DESCRIPTION
**Pull request by jaronoff97**
**Description:** 
Release 0.102.0. This contains the patch release for the collector (0.102.1)

**Link to tracking Issue(s):** 
closes #3012 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,11 @@ AUTO_INSTRUMENTATION_NGINX_VERSION ?= "$(shell grep -v '\#' versions.txt | grep
 COMMON_LDFLAGS ?= -s -w
 OPERATOR_LDFLAGS ?= -X ${VERSION_PKG}.version=${VERSION} -X ${VERSION_PKG}.buildDate=${VERSION_DATE} -X ${VERSION_PKG}.otelCol=${OTELCOL_VERSION} -X ${VERSION_PKG}.targetAllocator=${TARGETALLOCATOR_VERSION} -X ${VERSION_PKG}.operatorOpAMPBridge=${OPERATOR_OPAMP_BRIDGE_VERSION} -X ${VERSION_PKG}.autoInstrumentationJava=${AUTO_INSTRUMENTATION_JAVA_VERSION} -X ${VERSION_PKG}.autoInstrumentationNodeJS=${AUTO_INSTRUMENTATION_NODEJS_VERSION} -X ${VERSION_PKG}.autoInstrumentationPython=${AUTO_INSTRUMENTATION_PYTHON_VERSION} -X ${VERSION_PKG}.autoInstrumentationDotNet=${AUTO_INSTRUMENTATION_DOTNET_VERSION} -X ${VERSION_PKG}.autoInstrumentationGo=${AUTO_INSTRUMENTATION_GO_VERSION} -X ${VERSION_PKG}.autoInstrumentationApacheHttpd=${AUTO_INSTRUMENTATION_APACHE_HTTPD_VERSION} -X ${VERSION_PKG}.autoInstrumentationNginx=${AUTO_INSTRUMENTATION_NGINX_VERSION}
 ARCH ?= $(shell go env GOARCH)
+ifeq ($(shell uname), Darwin)
+  SED_INPLACE := sed -i ''
+else
+  SED_INPLACE := sed -i
+endif
```
Comment (jaronoff97) [inline Makefile:23]: This was preventing me from running the make release for the operator... cc @swiatekm-sumo let me know if this is horribly wrong.

File: Makefile [hunk 2]
```diff
@@ -527,14 +532,14 @@ chlog-insert-components:
 	@echo "* [OpenTelemetry Collector - v${OTELCOL_VERSION}](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v${OTELCOL_VERSION})" >>components.md
 	@echo "* [OpenTelemetry Contrib - v${OTELCOL_VERSION}](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v${OTELCOL_VERSION})" >>components.md
 	@echo "* [Java auto-instrumentation - v${AUTO_INSTRUMENTATION_JAVA_VERSION}](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v${AUTO_INSTRUMENTATION_JAVA_VERSION})" >>components.md
-	@echo "* [.NET auto-instrumentation - v${AUTO_INSTRUMENTATION_DOTNET_VERSION}](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/tag/$v{AUTO_INSTRUMENTATION_DOTNET_VERSION})" >>components.md
+	@echo "* [.NET auto-instrumentation - v${AUTO_INSTRUMENTATION_DOTNET_VERSION}](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/tag/v${AUTO_INSTRUMENTATION_DOTNET_VERSION})" >>components.md
```
Comment (jaronoff97) [inline Makefile:535]: this also was borked

File: config/manager/kustomization.yaml [hunk 3]
```diff
@@ -1,2 +1,8 @@
 resources:
 - manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+images:
+- name: controller
+  newName: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator
+  newTag: 0.102.0
```
Comment (TylerHelmuth) [inline config/manager/kustomization.yaml]: Classic prep-pr issue

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,42 @@
...
+
+- `collector`: Fixes a bug that was preventing regexes from being loaded correctly. Now the filter provide is exactly what's used. (#3007)
+  This is technically a breaking change if a user relied on the previously broken regex functionality.
+  This change will actually fix their regex to work where it didn't before. I expect that users would rather their
+  regexes work than break silently.
+  
+
+### Components
+
+* [OpenTelemetry Collector - v0.102.1](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.102.1)
```
Comment (TylerHelmuth) [inline CHANGELOG.md:31]: It is probably worth calling out that this version fixes the CVE and linking to it.

File: versions.txt [hunk 5]
```diff
@@ -23,11 +23,11 @@ autoinstrumentation-nodejs=0.51.0
 
 # Represents the current release of Python instrumentation.
 # Should match value in autoinstrumentation/python/requirements.txt
-autoinstrumentation-python=0.44b0
+autoinstrumentation-python=0.45b0
 
 # Represents the current release of DotNet instrumentation.
 # Should match autoinstrumentation/dotnet/version.txt
-autoinstrumentation-dotnet=1.2.0
+autoinstrumentation-dotnet=1.6.0
```
Comment (TylerHelmuth) [inline versions.txt:30]: No don't do this yet, this is the version with breaking changes to http semantic conventions, and we still havent decided when to take that hit by default

**Review by TylerHelmuth — CHANGES_REQUESTED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
go instrumentation just released https://github.com/open-telemetry/opentelemetry-go-instrumentation/releases/tag/v0.13.0-alpha if you want to get that in

**Review by pavolloffay — APPROVED** [review]

File: RELEASE.md [hunk 6]
```diff
@@ -44,9 +44,9 @@ The operator should be released within a week after the [OpenTelemetry collector
...
 |----------|-----------------|
-| v0.102.0 | @swiatekm-sumo  |
 | v0.103.0 | @frzifus        |
 | v0.104.0 | @jaronoff97     |
 | v0.105.0 | @pavolloffay    |
 | v0.106.0 | @yuriolisa      |
-| v0.107.0 | @TylerHelmuth   |
\ No newline at end of file
+| v0.107.0 | @TylerHelmuth   |
+| v0.102.0 | @swiatekm-sumo  |
```
Comment (swiatekm-sumo) [inline RELEASE.md:52]: If you're going to steal this release from me, you should keep me at the top of the list and move yourself to the bottom imo.

File: Makefile (hunk 1 above)
Comment (swiatekm-sumo) [reply Makefile:23]: It's probably better in the long run to just require `gsed` on Mac, but this will do for now.

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**